### ISSUE TITLE
Simplify accessing a repository

### ISSUE THREAD

**Issue by hagenw** [audeering/audbackend#197]
After the introduction of the interfaces, we removed the return value of `audbackend.create()` (https://github.com/audeering/audbackend/pull/174) after [this discussion](https://github.com/audeering/audbackend/pull/171#issuecomment-1909725543).
I think this was a good idea, but I would even go further and propose that `audbackend.access()` is no longer needed.

To access a backend, you can do at the moment:
```python
interface = audbackend.access(backend, host, repo)
```
It will return the default versioned interface, but a user might not be aware of this.

To access a different interface, that might even have arguments:
```python
interface = audbackend.access(
    backend,
    host,
    repo,
    interface=audbackend.core.FancyInterface,
    interface_kwargs={"myarg": 0},
)
```
This feels a little bit cumbersome and also strange as it returns an interface, and you have to provide an interface as argument.

I would propose instead to remove (deprecate) `audbackend.access()` and change how an interface is initialized to:
```python
interface = audbackend.core.FancyInterface(backend, host, repo, myarg=0)
```
Then it will be always obvious which interface will be returned, and a user needs only to look up the documentation of the interface, not of the interface and `audbackend.access()` together.

**Comment by frankenjoe** [comment]
I stopped using `audbackend.access()` already some time ago, instead in my projects I simply do:

``` python
backend = audbackend.MyBackend(host, repo)
interface = audbackend.interface.MyInterface(backend)
```

But maybe we should think about also deprecating `create()`, `delete()` and `register()` then and instead introduce `audbackend.Base.create()` and `audbackend.Base.delete()`?

**Comment by frankenjoe** [comment]
Otherwise, if we decide to keep the API functions. Then I would propose we deprecate `audbackend.access()` and instead introduce a new function like `audbackend.backend()` that works like `access()` but returns `audbackend.backend.Base`. Then a user would e.g. do:

```python
backend = audbackend.backend(name, host, repo)
interface = audbackend.interface.Versioned(backend)
```

**Comment by frankenjoe** [comment]
btw: if we deprecate `access()` we can immediately deprecate `interface.Unversioned`, too :)

**Comment by hagenw** [comment]
One advantage (or maybe the motivation) of `create()`, `delete()`, and `register()` are that a user does not need to deal with the backend classes. I had this in mind when I proposed to use `audbackend.interface.MyInterface(name, host, repo)` to replace `access()` as it would be in line with that approach.

But good that you pointed out that there is indeed the other solution, which would **get rid of backend names**.
To access an existing backend:

```python
backend = audbackend.backend.MyBackend(host, repo)
interface = audbackend.interface.MyInterface(backend)
```

We could easily add a `delete()` method here.
Adding a way to create the backend is slightly more complicated. One way might be to use a class method:

```python
backend = audbackend.backend.MyBackend.create(host, repo)
```

We would remove the following functions with this approach:
* `audbackend.access()`
* `audbackend.create()`
* `audbackend.delete()`
* `audbackend.register()`

---

I'm not completely decided yet, but the second approach of getting rid of backend names seems a little bit more straightforward to me. Any opinion?

**Comment by hagenw** [comment]
> btw: if we deprecate `access()` we can immediately deprecate `interface.Unversioned`, too :)

I see the point, as the functionality is already given by the backend. But from a user interface perspective it feels strange, as a it is easier to advice a user to always wrap an interface around a backend. This was also the main idea behind the approach to hide the backend classes from the user.

The other disadvantage is that stuff like `isinstance(obj, audbackend.interface.Base)` will then not work, if the user uses directly the backend.

But I guess, there is also no way to move more functionality from the backend class to the base interface class, to avoid that `audbackend.interface.Unversioned` adds no functionality to a backend?

**Comment by frankenjoe** [comment]
> I'm not completely decided yet, but the second approach of getting rid of backend names seems a little bit more straightforward to me. Any opinion?

Yes, seems unnecessarily complicated to first register a class and afterwards use the backend name to create an instance, when a user can also directly instantiate that instance. It's only convenient when working with config files, but that's not something we need to deal within the `audbackend` package. So I would vote for deprecating those api functions.

---

> Adding a way to create the backend is slightly more complicated. 

We can use a class method as you propose, though it feels a little strange to me that `create()` is a class method and `delete()` is not. The advantage of a class method is of course that then we don't an additional function like `audbackend.backend.Base.access()` to access a (existing) repository. 

On the other hand it is very common to open and close connections to a storage drive. So maybe we should actually introduce both, e.g.:

```python
backend = audbackend.backend.MyBackend(host, repo)
backend.open()
# do something with the backend
backend.close()
```

or alternatively:

```python
with audbackend.backend.MyBackend(host, repo) as backend:
     # do something with the backend
```

**Comment by hagenw** [comment]
Great, to summarize the proposal related to this issue:

Deprecate the following functions:

* `audbackend.access()` -> use a backend class directly
* `audbackend.create()` -> use the class method `audbackend.backend.Backend.create()`
* `audbackend.delete()` -> use the method `audbackend.backend.Backend.delete()`
* `audbackend.register()` -> handle inside packages that need names for config files, e.g. `audb`

Add the following functionality:

* `backend = audbackend.backend.Backend.create()` class method to create a repository on a backend
* `audbackend.backend.Backend.delete()` (class?) method to delete a backend

---

For `open()`, `close()` I created https://github.com/audeering/audbackend/issues/199

**Comment by frankenjoe** [comment]
> audbackend.backend.Backend.delete() (class?) method to delete a backend

If we make `create()` a class function, we should do the same with `delete()` I think. It also avoids that a user does the following:

```python
with MyBackend(host, repo) as backend:
    backend.delete()
    # now `backend.close()` gets automatically called on a non-existing repo
```

**Comment by hagenw** [comment]
Solved by #200, #202, #203.
